types/scene/wlr_scene.c

```diff
@@ -1231,6 +1278,41 @@ static void scene_entry_render(struct render_list_entry *entry, const struct ren
 		};
 		fx_render_pass_add_rect(data->render_pass, &rect_options);
 		break;
+	case WLR_SCENE_NODE_SHADOW:;
+		struct wlr_scene_shadow *scene_shadow = wlr_scene_shadow_from_node(node);
+
+		struct wlr_box window_box;
+		int window_x = 0, window_y = 0, window_width = 0, window_height = 0, window_corner_radius = 0;
+		struct wlr_scene_node *parent_node = wl_container_of(node->link.prev, parent_node, link);
+		if (parent_node->type == WLR_SCENE_NODE_BUFFER || parent_node->type == WLR_SCENE_NODE_RECT) {
+			wlr_scene_node_coords(parent_node, &x, &y);
+			scene_node_get_size(node, &window_width, &window_height);
+			scene_node_get_corner_radius(node, &window_corner_radius);
+			printf("found window box %d, %d: width: %d, height %d\n", window_x, window_y, window_width, window_height);
+		}
+		window_box = (struct wlr_box) {
+		    .x = window_x,
+			.y = window_y,
+			.width = window_width,
+			.height = window_height
+		};
```

Comment:
We should probably also transform the box

Reply:
Yup, need to add that as a Todo before this one is ready